src/metax_api/api/rest/base/views/dataset_view.py

```diff
@@ -171,35 +171,33 @@ def get_all_urn_identifiers(self, request):
         urn_ids = [item['research_dataset']['urn_identifier'] for item in q]
         return Response(urn_ids)
 
-    def _search_using_other_dataset_identifiers(self):
+    def _search_using_dataset_identifiers(self):
         """
-        URN-lookup from self.lookup_field failed. Look from dataset json fields
-        preferred_identifier first, and array other_identifier after, if there are matches
+        Search by lookup value from urn_identifier and preferred_identifier fields. preferred_identifier
+        searched only with GET requests.
         """
-        lookup_value = self.kwargs.get(self.lookup_field)
-        obj = self._search_from_research_dataset({'urn_identifier': lookup_value}, False)
-
-        # allow preferred and other identifier searches only for GET, since their persistence
-        # is not guaranteed over time.
-        if not obj and self.request.method == 'GET':
-            obj = self._search_from_research_dataset({'preferred_identifier': lookup_value}, False)
-            if not obj:
-                obj = self._search_from_research_dataset(
-                    {'other_identifier': [{'local_identifier': lookup_value}]}, True)
-
-        if not obj:
-            raise Http404
-
-        return obj
+        lookup_value = self.kwargs.get(self.lookup_field, False)
 
-    def _search_from_research_dataset(self, search_json, raise_on_404):
         try:
             return super(DatasetViewSet, self).get_object(
-                search_params={'research_dataset__contains': search_json})
+                search_params={ 'research_dataset__contains': {'urn_identifier': lookup_value} })
         except Http404:
-            if raise_on_404:
+            if self.request.method != 'GET':
                 raise
-            return None
+
+        # search by preferred_identifier only for GET requests, while preferring:
+        # - hits from att catalogs (assumed to be first created. improve logic if situation changes)
+        # - latest version records (= has no next_version_id)
+        # - first created (the first harvested occurrence, probably)
```

Comment:
Created a ticket https://jira.eduuni.fi/browse/CSCMETAX-316 for a thought that crossed my mind after reading this.